### PR TITLE
Stabilize group router spec

### PR DESCRIPTION
**Pull request by johanandren**
Resending if messages are lost. Also a general cleanup making it a bit easier to follow.

References #30671

### CODE REVIEW

File: akka-cluster-typed/src/test/scala/akka/cluster/typed/GroupRouterSpec.scala [hunk 1]
```diff
@@ -31,153 +35,166 @@ object GroupRouterSpec {
...
+            pongs += 1
+            uniqueWorkers += worker
+            if (pongs >= requestedPings) {
+              tellMeWhenDone ! DonePinging(pongs, uniqueWorkers)
+              Behaviors.stopped
+            } else {
+              Behaviors.same
+            }
+          case RetryPinging =>
+            // delivery isn't guaranteed so retry sending if we didn't get resonses in a timely fashion
```
Comment (octonato) [inline akka-cluster-typed/src/test/scala/akka/cluster/typed/GroupRouterSpec.scala:80]:
```suggestion
            // delivery isn't guaranteed so retry sending if we didn't get responses in a timely fashion
```

**Review by octonato — APPROVED** [review]
LGTM

(spotted a typo that I fixed myself)

**Comment by octonato** [comment]
failed because of https://github.com/akka/akka/issues/30664, I will restart the build

**Comment by octonato** [comment]
one more commit because of scalafmt

File: akka-cluster-typed/src/test/scala/akka/cluster/typed/GroupRouterSpec.scala [hunk 2]
```diff
@@ -31,154 +35,165 @@ object GroupRouterSpec {
...
+        }
+        val router = ctx.spawn(groupRouter, "group-router")
+        ctx.system.receptionist ! Receptionist.Subscribe(pingPongKey, ctx.self)
+
+        Behaviors.receiveMessage {
+          case pingPongKey.Listing(update) if update.size == settings.node1WorkerCount + settings.node2WorkerCount =>
+            // the requested number of workers are started and registered with the receptionist
+            // start pinging them
+            ctx.log.debug("Saw {} workers, starting pinger", update.size)
+            ctx.spawn(Pinger(router, settings.messageCount, resultProbe.ref), "pinger")
```
Comment (patriknw) [inline akka-cluster-typed/src/test/scala/akka/cluster/typed/GroupRouterSpec.scala:132]: If the setup is stable at this point, all workers registered and such, then I don't understand why we would need resend attempts in the Pinger? Messages are not lost under normal conditions.

Comment (johanandren) [reply akka-cluster-typed/src/test/scala/akka/cluster/typed/GroupRouterSpec.scala:132]: I thought I saw `Upstream failed, cause: StreamTcpException` but now I see that is a different test (RemoteAskSpec) in the same log dump. Hmmm.

**Comment by johanandren** [comment]
Removed the retrying but figured out a potential race condition that is now fixed:

The test would create a router, subscribe to the receptionist and only start pinging once it saw an update from the receptionist with the right number of workers. This did not guarantee the router had seen the right number of workers yet. Changed to create the router only after observing the number of workers so that it will see all from start (first listing it gets from receptionist).

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**